impeller/compiler/code_gen_template.h

```diff
@@ -50,7 +50,7 @@ struct {{camel_case(shader_name)}}{{camel_case(shader_stage)}}Shader {
 {% for def in struct_definitions %}
   struct {{def.name}} {
 {% for member in def.members %}
-    {{member.type}} {{member.name}}; // (offset {{member.offset}}, size {{member.byte_length}})
+{% if member.element_padding > 0 %}Padded<{{member.type}}, {{member.element_padding}}>{% else %}{{member.type}}{% endif %} {{" " + member.name}}{% if member.array_elements > 1 %}[{{member.array_elements}}]{% endif %}; // (offset {{member.offset}}, size {{member.byte_length}})
```

Comment:
BTW?, how is the templating in code generation working out? I added it initially because I was losing my mind concatenating strings and thought this was easier to read (and also I had the JSON file with just the data separately that I could use a quick reference for what was available). But the templates themselves are not particularly great either :/

Reply:
TBH I love the design, but it could be a bit of lingering Stockholm syndrome from my Ansible and SaltStack days. This jinja2 implementation seems to be missing [whitespace management](https://jinja.palletsprojects.com/en/3.1.x/templates/#whitespace-control) features that I'm used to like `{%-`, which would help make this part of the template easier to read without messing up the output.

Reply:
I believe that already works. I just set it to [on by default here](https://github.com/flutter/engine/blob/fb067eab70c5b9f36d152c5501bf263c4254f1df/impeller/compiler/reflector.cc#L405). We could turn it off and manage whitespace manually like you're used to. Haven't used Ansible or Saltstack so I can't relate to that experience (perhaps for the best?).

Reply:
We could also just pipe it through clang-format in the GN template and not worry about whitespace at all.

Reply:
Ah that explains a lot. I think I'll swing back around and play with the whitespace a bit more later to get this line more readable.